test/test_rake_task_argument_parsing.rb

```diff
@@ -32,8 +32,8 @@ def test_two_arguments
     assert_equal ["one", "two"], args
   end
 
-  def test_can_handle_spaces_between_args
-    name, args = @app.parse_task_string("name[one, two,\tthree , \tfour]")
+  def test_can_handle_spaces_between_all_args
+    name, args = @app.parse_task_string("name[ one , two ,\tthree , \tfour ]")
```

Comment:
Keep the existing test and add a new test as this is technically a new feature. Also, make sure it doesn't strip out whitespaces in a string literal:

```sh
rake name[one,two,three]     # => should be interpreted as `rake name[one,two,three]`
rake name[ one , two, three] # => should be interpreted as `rake name[one,two,three]`

rake name["one two",three]   # => should be interpreted as `rake name["one two",three]`
rake name[ "one two", three] # => should be interpreted as `rake name["one two",three]`
```

Reply:
Thanks for the feedback.

Added original test back.
Regarding your second comment, i think there's already a test case below testing that whitespaces embedded in arguments are not strip out. Let me know if that covers it.

Added new commit, let me know if you prefer amend.